### PR TITLE
Update Spectacle types to reflect latest API

### PR DESCRIPTION
**Pull request by kale-stew**
This PR updates the Spectacle type definitions to match the latest API.

Closes https://github.com/FormidableLabs/spectacle/issues/556.

---

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/FormidableLabs/spectacle/releases _(bumping from v1.2.0 to v5.2.2)_
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. _(one was already initialized)_

### CODE REVIEW

**Comment by ebrillhart** [comment]
I'm the current Lead Maintainer for Spectacle and I approve this PR if that's helpful context! This has been highly requested.

**Comment by zmaybury** [comment]
Nice work, these types were woefully out of date! (I originally wrote them for 1.2 as none existed at the time). Approved!

**Review by zmaybury — APPROVED** [review]
Awesome work, thanks for updating!

**Pull request merged**
